readmes/README.macOS.md

```diff
@@ -1,153 +1,140 @@
-PermitUserEnvironment yes
-EOF
-```
+BREWPREFIX=$(brew --prefix); export PATH="$BREWPREFIX/opt/gnu-sed/libexec/gnubin:$BREWPREFIX/opt/apr/bin:$PATH"; CXXFLAGS="-I $BREWPREFIX/include" CFLAGS="-ggdb -Og -g3 -fno-omit-frame-pointer -I $BREWPREFIX/include" LDFLAGS="-L $BREWPREFIX/lib" CC=$(which gcc-13) CXX=$(which g++-13) ./configure --enable-debug --prefix=$(cd ~; pwd)/install/cbdb;
 
-## 3. Enable ssh to localhost without requiring a password
+# 2. Compile and install Cloudberry Database.
 
-Currently, the GPDB utilities require that it be possible to `ssh` to localhost
-without using a password.  To verify this, the following command should succeed
-without erroring, or requiring you to enter a password. 
+make -j8
+make -j8 install
 
-```bash
-ssh <hostname of your machine>  # e.g., ssh briarwood (You can use `hostname` to get the hostname of your machine.)
-```
+# 3. Bring in Greenplum environment for Cloudberry Database into your running shell.
 
-However, if it is the first time that you are `ssh`'ing to localhost, you may
-need to accept the trust on first use prompt.
+source $(cd ~; pwd)/install/cbdb/greenplum_path.sh
 
-```bash
-The authenticity of host '<your hostname>' can't be established.
-ECDSA key fingerprint is SHA256:<fingerprint here>.
-Are you sure you want to continue connecting (yes/no)?
-```
+# 4. Install the Python dependencies.
 
-If the hostname does not resolve, try adding your machine name to `/etc/hosts`,
-like so:
+pip3 install --user -r readmes/python-dependencies.txt
 
-```bash
-echo -e "127.0.0.1\t$HOSTNAME" | sudo tee -a /etc/hosts
-```
-
-If you see `ssh: connect to host <> port 22: Connection refused`, enable remote
-login in:
-
-```bash
-System Preferences -> Sharing -> Remote Login
-```
+# 5. Start a demo cluster.
 
-If you see a password prompt, add your SSH key to the `authorized_keys` file,
-like so:
+PORT_BASE=8000 make create-demo-cluster
 
-```bash
-mkdir -p ~/.ssh
-ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
-cat ~/.ssh/id_rsa.pub >>  ~/.ssh/authorized_keys
+# (gpdemo-env.sh contains __PGPORT__ and __MASTER_DATA_DIRECTORY__ values)
 ```
+## Step 4. Verify the cluster
+
+1. You can verify whether the cluster has started successfully by running the following command. You will see many active `postgres` processes with ports ranging from `8000` to `8007`.
+
+    ```bash
+    ps -ef | grep postgres
+    ```
+    
+2. Connect to the Cloudberry Database and see the active segment information by querying the system table `gp_segement_configuration`. For detailed description of this table, see the Greenplum document [here](https://docs.vmware.com/en/VMware-Greenplum/6/greenplum-database/ref_guide-system_catalogs-gp_segment_configuration.html).
+
+    ```sql
+    $ psql -p 8000 postgres
+    postgres=# select version();
+    postgres=# select * from gp_segment_configuration;
+    ```
+    
+    Example output:
+
+    ```shell
+    hashdatas-MacBook-Pro:cloudberrydb hashdata $ psql -p 8000 postgres
```

Comment:
we can delete the specific hostname info `hashdatas-MacBook-Pro:cloudberrydb hashdata`

Reply:
FYI, if we need to change hostname, the  `hostname` and `address` columns in the psql output should be the same as the machine hostname. we'd better change them all.

Reply:
Thank you for your suggestion. I'll revise it